src/main/scala/com/lunatech/pointingpoker/actors/Room.scala

```diff
@@ -0,0 +1,116 @@
+
+  sealed trait Response
+  final case class DataStatus(data: RoomData) extends Response
+
+  final case class User(id: UUID, name: String, voted: Boolean, estimation: String, ref: UntypedRef)
+
+  final case class RoomData(users: List[User], currentIssue: String, issueLastEditBy: Option[UUID])
+  object RoomData {
+    val empty: RoomData = RoomData(List.empty[User], "", Option.empty[UUID])
+  }
+
+  def apply(roomId: UUID): Behavior[Command] =
+    Behaviors.setup[Command] { _ =>
+      receiveBehaviour(roomId, RoomData.empty)
+    }
+
+  private[actors] def receiveBehaviour(roomId: UUID, data: RoomData): Behavior[Command] =
+    Behaviors.receive[Command] { (context, message) =>
+      message match {
+        case Join(user) =>
+          val newData = data.copy(users = user :: data.users)
+          setupNewUser(user, roomId, newData)
+          broadcast(WSMessage(MessageType.Join, roomId, user.id, user.name), newData.users, context)
+          receiveBehaviour(roomId, newData)
+        case Vote(userId, estimation) =>
+          val newUsers = data.users.map { u =>
+            if (userId == u.id) u.copy(voted = true, estimation = estimation)
+            else u
+          }
+          broadcast(WSMessage(MessageType.Vote, roomId, userId, estimation), newUsers, context)
+          receiveBehaviour(roomId, data.copy(users = newUsers))
+        case ClearVotes(userId) =>
+          val newUsers = data.users.map(_.copy(voted = false, estimation = ""))
+          broadcast(
+            WSMessage(MessageType.Clear, roomId, userId, WSMessage.NoExtra),
+            newUsers,
+            context
+          )
+          receiveBehaviour(roomId, data.copy(users = newUsers))
+        case ShowVotes(userId) =>
+          broadcast(
+            WSMessage(MessageType.Show, roomId, userId, WSMessage.NoExtra),
+            data.users,
+            context
+          )
+          Behaviors.same
+        case Leave(userId) =>
+          val newUsers = data.users.filter(_.id != userId)
+          broadcast(
+            WSMessage(MessageType.Leave, roomId, userId, WSMessage.NoExtra),
+            newUsers,
+            context
+          )
+          if (newUsers.isEmpty) {
+            Behaviors.stopped
+          } else {
+            receiveBehaviour(roomId, data.copy(users = newUsers))
+          }
+        case EditIssue(userId, issue) =>
+          broadcast(WSMessage(MessageType.EditIssue, roomId, userId, issue), data.users, context)
+          receiveBehaviour(
+            roomId,
+            data.copy(currentIssue = issue, issueLastEditBy = Option(userId))
+          )
+        case GetData(replyTo) =>
+          replyTo ! Room.DataStatus(data)
+          Behaviors.same
+      }
+
+    }
+
+  private[actors] def broadcast(
+      message: WSMessage,
+      users: List[User],
+      context: ActorContext[Command]
+  ): Unit = {
+    context.log.info("Broadcasting: {} ", message)
```

Comment:
Should this be a debug-level log instead? I would expect a lot of broadcasts to be happening (and flooding the logs).

Reply:
Done =]